src/cpp/session/resources/terminal/bash/.bash_profile

```diff
@@ -5,12 +5,15 @@ _FAKEHOME="${HOME}"
 HOME="${_REALHOME}"
 
 # source the user startup file, if any
+STARTUP_FILE="/etc/profile.d/*.sh"
 if [ -f ~/.bash_profile ]; then
 	source ~/.bash_profile
 elif [ -f ~/.bash_login ]; then
 	source ~/.bash_login
 elif [ -f ~/.profile ]; then
 	source ~/.profile
+elif ls $STARTUP_FILE 1> /dev/null 2>&1; then
```

Comment:
Iterating over the output of `ls` is dangerous; instead prefer using something like:

```
for PROFILE in /etc/profile.d/*.sh; do
  if [ -e "${PROFILE}" ]; then
    source "${PROFILE}"
  fi
done
```

See also: https://superuser.com/a/418490/317257

Reply:
Good point, thanks! Any particular reason to use `-e` over `-f` here?

Reply:
Good point -- this should probably be `-f` since you can't source a directory. 🙂 Thanks!